src/libstd/dynamic_lib.rs

```diff
@@ -263,6 +266,16 @@ mod dl {
                  symbol: *const libc::c_char) -> *mut libc::c_void;
         fn dlclose(handle: *mut libc::c_void) -> libc::c_int;
     }
+
+    #[cfg(target_os = "openbsd")]
```

Comment:
The extern block below is missing a `#[link_name]` attribute. I'm not sure what the purpose of `#[link_name]` on an extern block is (I think it should be `#[link(name="dl"]` here?).